cmd/devp2p/internal/ethtest/transaction.go

```diff
@@ -66,31 +66,42 @@ func sendSuccessfulTxWithConn(t *utesting.T, s *Suite, tx *types.Transaction, se
 	}
 }
 
-func sendFailingTx(t *utesting.T, s *Suite, tx *types.Transaction) {
+func sendFailingTxs(t *utesting.T, s *Suite, txs []*types.Transaction) {
 	sendConn, recvConn := s.setupConnection(t), s.setupConnection(t)
 	defer sendConn.Close()
 	defer recvConn.Close()
-	sendFailingTxWithConns(t, s, tx, sendConn, recvConn)
+	sendFailingTxsWithConns(t, s, txs, sendConn, recvConn)
 }
 
-func sendFailingTxWithConns(t *utesting.T, s *Suite, tx *types.Transaction, sendConn, recvConn *Conn) {
-	// Wait for a transaction announcement
-	switch msg := recvConn.ReadAndServe(s.chain, timeout).(type) {
-	case *NewPooledTransactionHashes:
-		break
-	default:
-		t.Logf("unexpected message, logging: %v", pretty.Sdump(msg))
-	}
+func sendFailingTxsWithConns(t *utesting.T, s *Suite, txs []*types.Transaction, sendConn, recvConn *Conn) {
 	// Send the transaction
-	if err := sendConn.Write(&Transactions{tx}); err != nil {
+	txMsg := Transactions(txs)
+	if err := sendConn.Write(&txMsg); err != nil {
```

Comment:
I think we should announce the txs individually here. Some of them can be large, and the node might reject the entire announcement when the first item is too big.

Reply:
What I meant is, you can just send them in a loop like

```
for _, tx := range txs {
    txMsg := Transactions(txs)
    if err := sendConn.Write(&txMsg); err != nil { ... }
}
```